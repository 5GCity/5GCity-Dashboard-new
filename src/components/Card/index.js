/**
 * Card Component
 * Please write a description
 *
 * @author Guilherme Patriarca <gpatriarca@ubiwhere.com>
 */
import styled from 'styled-components'
import { Card } from 'element-react';

export default styled(Card)`
    border-radius: 6px;
    background-color: #48565E;
    box-shadow: 0 0 20px 0 rgba(0,0,0,0.1);
    border:0;

    .el-card__header {
        border-bottom: 1px solid rgba(137,151,159,0.2);
      }
    
`
