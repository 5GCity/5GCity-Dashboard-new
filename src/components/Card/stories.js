/**
 * Card Component Stories
 * Please write a description or remove this line
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import { storiesOf } from '@storybook/react'
import { Theme } from 'globalStyles'
import Card from './index'

const bStyle = {
  padding: '30px',
  backgroundColor: Theme.bodyBackground,
  height: '100%'
}

const example1 = () => (
  <div style={bStyle}>
    <Card>
      <div className='text item'>List item 1</div>
      <div className='text item'>List item 2</div>
      <div className='text item'>List item 3</div>
      <div className='text item'>List item 4</div>
    </Card>
  </div>
)

const example2 = () => (
  <div style={bStyle}>
    <Card
      header={
        <div>
          <span>Card Name</span>
        </div>
  }
>
      <div className='text item'>List item 1</div>
      <div className='text item'>List item 2</div>
      <div className='text item'>List item 3</div>
      <div className='text item'>List item 4</div>
    </Card>
  </div>
)

storiesOf('Card', module)
  .add('default', example1)
  .add('default + header', example2)
