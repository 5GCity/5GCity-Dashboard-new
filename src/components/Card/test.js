/**
 * Card Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import Card from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<Card />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
