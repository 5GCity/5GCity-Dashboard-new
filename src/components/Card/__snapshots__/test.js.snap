// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  border-radius: 6px;
  background-color: #48565E;
  box-shadow: 0 0 20px 0 rgba(0,0,0,0.1);
  border: 0;
}

.c0 .el-card__header {
  border-bottom: 1px solid rgba(137,151,159,0.2);
}

<div
  className="el-card c0"
  style={Object {}}
>
  <div
    className="el-card__body"
    style={
      Object {
        "padding": "20px",
      }
    }
  />
</div>
`;
